ex_cubic_ingestion/lib/ex_cubic_ingestion/validators.ex

```diff
@@ -24,4 +24,10 @@ defmodule ExCubicIngestion.Validators do
   def map_has_keys?(map, key_list) do
     Enum.all?(key_list, &Map.has_key?(map, &1))
   end
+
+  @spec valid_s3_object?(map()) :: boolean()
+  def valid_s3_object?(object) do
+    Map.has_key?(object, :key) && not String.ends_with?(object.key, "/") &&
+      Map.has_key?(object, :size) && object.size != "0"
+  end
```

Comment:
suggestion: this might be better as a pattern match
```suggestion
  def valid_s3_object?(object)
  def valid_s3_object?(%{key: key, size: size}) when size != "0" do
    not String.ends_with?(key, "/")
  end
  def valid_s3_object?(%{}) do
    false
  end
```

Reply:
Fixed.